packages/integration_test/lib/common.dart

```diff
@@ -114,16 +114,20 @@ class Failure {
   /// The name of the test method which failed.
   final String methodName;
 
+  /// The error that was thrown during the test.
+  final Object error;
+
   /// The details of the failure such as stack trace.
   final String details;
 
   /// Constructor requiring all fields during initialization.
-  Failure(this.methodName, this.details);
+  Failure(this.methodName, this.details, {this.error});
 
   /// Serializes the object to JSON.
   String toJson() {
     return json.encode(<String, String>{
       'methodName': methodName,
+      'error': error.toString(),
```

Comment:
I think we probably want this to be
```suggestion
      'error': error != null ? error.toString() : null,
```

To avoid having the string literally `'null`' in here.

Reply:
```suggestion
      'error': error?.toString(),
```

Reply:
@natebosch - that would have the unfortunate side effect of emitting `'null`' where we probably want `null`.

Reply:
It shouldn't. `'$error'` would do that, but `error?.toString()` should stay `null` and not `'null'`.

Reply:
Ahhh you are right.